### PR TITLE
Fix generic type handling

### PR DESCRIPTION
**Pull request by chrisdunelm**

### CODE REVIEW

File: tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs [hunk 1]
```diff
@@ -30,6 +30,18 @@ private static string Show(this IList<GenericParameter> ps)
             return "";
         }
 
+        private static string ShowGenArgs(this IList<TypeReference> types)
```
Comment (jskeet) [inline tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs]: Any reason for not using the conditional ?: operator and an expression bodied method here?

File: tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs [hunk 2]
```diff
@@ -43,13 +55,26 @@ private static string Show(this ParameterDefinition p)
...
                 sb.Append(" = ");
-                sb.Append(p.Constant ?? "null"); // TODO: I suspect `Constant` is not thw right way to do this.
+                sb.Append(p.Constant ?? "null"); // TODO: I suspect `Constant` is not the right way to do this.
             }
             return sb.ToString();
         }
 
-        public static string Show(this TypeReference type) =>
-            type.FullName == "System.Void" ? "void" : $"{type.FullName.Split('`')[0].Replace('/', '.')}{type.GenericParameters.Show()}";
+        public static string Show(this TypeReference type)
```
Comment (jskeet) [inline tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs]: As a nit, I think Format would be a better name than Show. But that's not really a matter of this PR.

File: tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs (hunk 1 above)
Comment (chrisdunelm) [reply tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs]: No reason, done :)

File: tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs (hunk 2 above)
Comment (chrisdunelm) [reply tools/Google.Cloud.Tools.VersionCompat/CecilUtils/ShowExtensions.cs]: ack

**Pull request merged**